src/core/lbgpu.cpp

```diff
@@ -440,4 +442,33 @@ int lb_lbfluid_load_checkpoint_wrapper(char* filename, int binary)
   return lb_lbfluid_load_checkpoint(filename, binary);
 }
 
+void lb_lbfluid_particles_add_momentum ( float velocity[3] )
+{
+  for (int i = 0; i < lbpar_gpu.number_of_particles; ++i)
+  {
+    double mass;
+#ifdef MASS
+    mass = particle_data_host[i].mass;
+#else
+    mass = 1.;
+#endif
+
+    double new_velocity[3] = {
+      particle_data_host[i].v[0] + velocity[0] / mass * time_step,
```

Comment:
That’s also in the interface.  From `particle_data.pyx`:
````python
myv[i] *= time_step
````
The parameter `velocity` is the one returned from `calc_linear_momentum` which is already scaled back to interface units, so we have to multiply by time step again.  Trust me, I checked this.